Automattic/wp-calypso
# Add StatsUpsellModal

Fixes https://github.com/Automattic/dotcom-forge/issues/4911

cc @crisbusquets 

## Proposed Changes

* Adds the StatsUpsellModal

## Testing Instructions

* Visit the calypso live link, append the query param to the url enable the paywall gating for the session:  `?flags=stats/paid-wpcom-v2`
* e.g. https://container-nifty-newton.calypso.live/?flags=stats/paid-wpcom-v2
* Click on a stats specific upgrade button under /stats

Todo:
- [x] Confirm we're still going with a personal upsell
- [x] Fix plan pricing and localization
- [x] Localized currency font size needs reducing to match figma
- [x] Check tracks events are firing right
- [x] Check checkout link redirect to works and ideally update it to go back to the current path instead of just /stats/day/:site

Doing in follow up prs:
- [ ] Use it for the download csv button
- [ ] Use it for the banner button
- [ ] If using premium, account for upgrades from personal in the pricing being displayed - moved: https://github.com/Automattic/dotcom-forge/issues/4997

![Screenshot 2023-12-19 at 17-05-08 Jetpack Stats ‹ Site Title — WordPress com](https://github.com/Automattic/wp-calypso/assets/811776/f06e6006-17a9-4b30-a26a-ecd823320b51)
